ow_faults_detection/src/FaultDetector.cpp

```diff
@@ -8,10 +8,95 @@
 using namespace std;
 using namespace ow_lander;
 
+constexpr std::bitset<10> FaultDetector::isCamExecutionError;
+constexpr std::bitset<10> FaultDetector::isPanTiltExecutionError;
+constexpr std::bitset<10> FaultDetector::isArmExecutionError;
+constexpr std::bitset<10> FaultDetector::isPowerSystemFault;
+
+constexpr std::bitset<3> FaultDetector::islowVoltageError;
+constexpr std::bitset<3> FaultDetector::isCapLossError;
+constexpr std::bitset<3> FaultDetector::isThermalError;
+
 FaultDetector::FaultDetector(ros::NodeHandle& node_handle)
 {
-  
   srand (static_cast <unsigned> (time(0)));
-  m_pub = node_handle.advertise<ow_faults::SystemFaults>("/test", 10);
+
+  // topics for JPL msgs: system fault messages, see Faults.msg, Arm.msg, Power.msg, PTFaults.msg
+  m_power_fault_msg_pub = node_handle.advertise<ow_faults::PowerFaults>("/faults/power_faults_status", 10);
+  m_system_fault_msg_pub = node_handle.advertise<ow_faults::SystemFaults>("/faults/system_faults_status", 10);
+
+  //  power fault publishers and subs
+  m_power_soc_sub = node_handle.subscribe("/power_system_node/state_of_charge",
+                                          10,
+                                          &FaultDetector::powerSOCListener,
+                                          this);
+  m_power_temperature_sub = node_handle.subscribe("/power_system_node/battery_temperature",
+                                                  10,
+                                                  &FaultDetector::powerTemperatureListener,
+                                                  this);
+}
+
+// Creating Fault Messages
+template<typename fault_msg>
+void FaultDetector::setFaultsMessageHeader(fault_msg& msg){
+  msg.header.stamp = ros::Time::now();
+  msg.header.frame_id = "world";
 }
 
+template<typename bitsetFaultsMsg, typename bitmask>
+void FaultDetector::setBitsetFaultsMessage(bitsetFaultsMsg& msg, bitmask bm) {
+  setFaultsMessageHeader(msg);
+  msg.value = bm.to_ullong();
+}
+
+template<typename fault_msg>
+void FaultDetector::setComponentFaultsMessage(fault_msg& msg, ComponentFaults value) {
+  setFaultsMessageHeader(msg);
+  msg.value = static_cast<uint>(value);
+}
+
+float FaultDetector::getRandomFloatFromRange( float min_val, float max_val){
+  return min_val + (max_val - min_val) * (rand() / static_cast<float>(RAND_MAX));
+}
```

Comment:
Formatting consistency
```suggestion
float FaultDetector::getRandomFloatFromRange(float min_val, float max_val) {
  return min_val + (max_val - min_val) * (rand() / static_cast<float>(RAND_MAX));
}
```